include/ur.py

```diff
@@ -196,6 +196,9 @@ class ur_function_v(IntEnum):
     ADAPTER_RETAIN = 179                            ## Enumerator for ::urAdapterRetain
     ADAPTER_GET_LAST_ERROR = 180                    ## Enumerator for ::urAdapterGetLastError
     ADAPTER_GET_INFO = 181                          ## Enumerator for ::urAdapterGetInfo
+    PROGRAM_BUILD_EXP = 197                         ## Enumerator for ::urProgramBuildExp
```

Comment:
dont we have all the changes in spec already merged in https://github.com/oneapi-src/unified-runtime/pull/924? I would expect this PR to only need to modify code in the L0 adapter, as the spec part has been already taken care of. Or is that this is not rebased?

Reply:
#924 was merged to main not adapters, we created #1083 to cherry-pick the spec changes into adapters since we were having trouble syncing adapters and main. I then made the decision to take the cherry-pick from #1083 into this PR to reduce the number of intel/llvm PR's required to get this feature merged in time for the next release.